plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -166,11 +169,29 @@ export async function createRouter(
 
   const workingDirectory = await getWorkingDirectory(config, logger);
   const integrations = ScmIntegrations.fromConfig(config);
-  let taskBroker: TaskBroker;
 
+  let taskBroker: TaskBroker;
   if (!options.taskBroker) {
     const databaseTaskStore = await DatabaseTaskStore.create({ database });
     taskBroker = new StorageTaskBroker(databaseTaskStore, logger);
+
+    if (scheduler && databaseTaskStore.listStaleTasks) {
+      await scheduler.scheduleTask({
+        id: 'close_stale_tasks',
+        frequency: { cron: '*/5 * * * *' }, // every 5 minutes, also supports Duration
+        timeout: { minutes: 15 },
+        fn: async () => {
+          const { tasks } = await databaseTaskStore.listStaleTasks({
+            timeoutS: 3600,
```

Comment:
Realizing that there's a non-0 change that someone might hit this limit tbh. Perhaps it's best to bump it up to something much safer, like 24h? Especially since the current implementation won't actually cancel ongoing tasks, meaning that the impact of shutting down an active task can be quite high due to the confusion.

Reply:
Makes sense, I've pushed the bump to `86400`